google/api_core/retry/retry_streaming.py

```diff
@@ -107,8 +107,9 @@ def retry_target_stream(
         time.monotonic() + timeout if timeout is not None else None
     )
     error_list: list[Exception] = []
+    sleep_iter = iter(sleep_generator)
 
-    for sleep in sleep_generator:
+    while True:
```

Comment:
Please could you file a follow up bug to refactor this code to add another exit condition? It's not ideal to use `while True` because there could be a scenario where the code is stuck in the while loop forever. Please could you add a comment clarifying the reason this loop will never be stuck indefinitely? If there is any doubt, we may need to block this PR until another exit condition is added. Otherwise, this is a non-blocking comment. Applies throughout.

Reply:
The logic should be the same as before. We've always relied on the wrapped function to tell us when to terminate (either by returning, or raise an exception)

When writing the code though, I did consider that now might be a good time to add a [max_attempts argument](https://github.com/googleapis/python-api-core/issues/535) as well, which would give us an exit condition here. Do you think I should consider implementing that as well? Or maybe adding a TODO referencing that it will come later?

Reply:
If there is no concern that there will be an infinite loop (eg. same behavior as before), a TODO comment will be sufficient. In terms of functional testing, I haven't been able to create a problem locally.

Reply:
sounds good, just added the comment